pkg/nodeconfig/nodeconfig.go

```diff
@@ -142,13 +144,28 @@ func getClusterAddr(kubeAPIServerEndpoint string) (string, error) {
 
 // Configure configures the Windows VM to make it a Windows worker node
 func (nc *nodeConfig) Configure() error {
+	drainHelper := &drain.Helper{Ctx: context.TODO(), Client: nc.k8sclientset}
+	// If we find a node  it implies that we are reconfiguring and we should cordon the node
+	if err := nc.setNode(true); err == nil {
+		// Make a best effort to cordon the node until it is fully configured
+		if err := drain.RunCordonOrUncordon(drainHelper, nc.node, true); err != nil {
+			nc.log.Info("unable to cordon", "node", nc.node.GetName(), "error", err)
+		}
+	}
+
 	if err := nc.Windows.Configure(); err != nil {
 		return errors.Wrap(err, "configuring the Windows VM failed")
 	}
-	// populate node object in nodeConfig
-	if err := nc.setNode(); err != nil {
+
```

Comment:
This is the point where a partially configured node is now present in the cluster. In order for Configure() to clean up properly before returning error, the rest of this function can be either be wrapped in an anonymous function, such as:
```
err := func() error {
	// all configuration steps after the Node object has been created
}()
if err != nil {
	// note: markNotReady is no longer exported
	if markErr := nc.markNotReady(); err != nil {
		nc.log.Info("Unable to mark node as NotReady", "error", markErr)
	}
	return err
}
return nil
```

Or alternatively it can be moved to a defined function.

Reply:
This looks cool. I will give it a shot.